File: Enwage/Models/Employee.cs
using System;
using System.Collections.Generic;

namespace Enwage.Models
{
    public partial class Employee
    {
        public Employee()
        {
            Attachments = new HashSet<Attachment>();
            Employeestates = new HashSet<Employeestate>();
        }

        public int Id { get; set; }
        public string Name { get; set; } = null!;
        public string Email { get; set; } = null!;
        public DateTime Dob { get; set; }
        public DateTime Experiencestart { get; set; }
        public DateTime Experienceend { get; set; }
        public string Yearsofexperience { get; set; } = null!;
        public decimal Rate { get; set; }
        public bool RateFlag { get; set; }
        public string? Gender { get; set; }
        public int ClientId { get; set; }

        public virtual Client Client { get; set; } = null!;
        public virtual ICollection<Attachment> Attachments { get; set; }
        public virtual ICollection<Employeestate> Employeestates { get; set; }
    }
}

File: Frontend/src/app/Models/Employee.ts
// src/app/models/employee.model.ts
export interface Employee {
    id?: number;
    name: string;
    email: string;
    dob: string; // Use string type for ISO date format
    experiencestart: string;
    experienceend: string;
    yearsOfExperience: string;
    rate: number;
    rateFlag: boolean;
    gender: string;
    clientId: number;
    employeestates: EmployeeState[];
    attachments?: Attachment[]; // Optional field
    noOfFiles: number; // Add this field
    noOfStates: number
  }
  
  export interface EmployeeState {
    id?: number;
    stateId: number;
    employeeId: number;
  }
  
  export interface Attachment {
    id?: number;
    name: string;
    size: number;
    type: string;
    employeeId: number;
    baseCode: string;
  }
  
  export interface GetAllEmployees {
    employees: Employee[];
    totalNumberOfRows: 0;
  }
   